src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    private static final List<FruitTransaction> dataStorage = new ArrayList<>();
```

Comment:
that's not a storage, but a transaction history record
storage should be a hashmap, with fruits as keys, their amounts as values - its role is to keep track of what fruits we have and how many

Reply:
You need to redo a big part of your logic - we don't rely on balance transactions to know how much we have but on the storage. Balance has its handlers, as any other operation type